swap/swap.go

```diff
@@ -72,39 +72,52 @@ type Owner struct {
 	publicKey  *ecdsa.PublicKey  // public key
 }
 
-// Params encapsulates param
+// Params encapsulates economic and operational parameters
 type Params struct {
+	OverlayAddr         []byte // this node's base address
```

Comment:
To be honest `Params` is poorly designed and was meant as an organizing thing for **economic** parametrization (thresholds, cashing strategy, amounts, etc.).

We have been refactoring the `Swap` constructor all the time recently, that is why instead of making a huuuuuge constructor with maaaany parameters (resulting in many changes), we started adding mixed parameters to this generic `Params` struct.

I would like to see a refactoring of this, but it does not need to be in this PR. Conceptually `OverlayAddr` does not belong to this struct, but `LogPath` neither does.

Reply:
i agree.

i never really understood this `Params` business—indeed, i only placed `OverlayAddr` here since it seemed to fit right next to `logpath` as an "infrastructure" sort of variable for the `Swap` struct :man_shrugging: 

we can have them moved to the `Swap` struct directly if you prefer, i don't mind.

or: we can revisit this issue later down the road, in a refactoring as you say

Reply:
created PR #1877 for this